lib/pbench/cli/agent/commands/tools/clear.py

```diff
@@ -0,0 +1,167 @@
+"""
+pbench-clear-tools
+
+This script will remove tools that have been registered.  If no options are
+used, then all tools from the "default" tool group are removed.  Specifying
+a tool name and/or remote host will limit the scope of the removal.
+
+
+"""
+import shutil
+import sys
+import os
+import pathlib
+
+import click
+
+from pbench.agent.utils import setup_logging
+from pbench.cli.agent import CliContext, pass_cli_context
+from pbench.cli.agent.commands.tools.base import ToolCommand
+from pbench.cli.agent.options import common_options
+
+
+class ClearTools(ToolCommand):
+    def __init__(self, context):
+        super().__init__(context)
+
+        self.logger = setup_logging(logfile=self.pbench_log)
+
+    def execute(self):
+        """Clear registered tools"""
+        self.tools_group_dir = self.tool_group_dir(self.context.group)
+        if not self.tools_group_dir.exists():
+            return 1
+
+        try:
+            self.context.remote = self.context.remote.split(",")
+        except Exception:
+            # We were not given any remotes on the command line, build the list
+            # from the tools group directory.
+            self.context.remote = self.remote(self.tools_group_dir)
+
+        for remote in self.context.remote:
+            tg_dir = self.tools_group_dir / remote
+            if not tg_dir.exists():
+                # Ignore any specified remote that does not exist.
+                self.logger.warn(
+                    f'The given remote host, "{remote}", is not a directory in {self.tools_group_dir}.'
+                )
+                continue
+
+            if self.context.name:
+                # Use the given tool name from the command line
+                self.context.name = [self.context.name]
+            else:
+                # Discovery all the tools registered for this remote
+                self.context.name = self.tools(tg_dir)
+
+            for name in self.context.name:
+                self._clear_tools(name, remote.name)
+
+            tool_files = [p.name for p in tg_dir.iterdir()]
+
+            if "__label__" in tool_files:
```

Comment:
Interestingly, the `bash` version deletes `__label__` only if it's the "last file standing" in the directory. Presumably it expects that `clear_tool` will have removed everything else. In fact, it's odd that the `clear_tool` function removes `__noinstall__` suffix files, but leaves `__label__` for the caller. Considering that we are apparently expecting nothing else to be left, the distinction between the script's `==` and your `in` probably doesn't matter.

Reply:
This still appears to be incorrect logic transferred from the original.  As Dave points out, we only want to remove `__label__` in this directory if we have no other tools left registered.

The `.__noinstall__` suffix files are per-registered tool, so they should be removed if that tool is removed.

Reply:
fixed